napari/layers/shapes/_shape_list.py

```diff
@@ -1136,17 +1150,24 @@ def inside(self, coord):
         if inside_indices.size == 0:
             return None
         try:
-            z_index = [self.shapes[i].z_index for i in inside_indices]
+            z_index = [
+                self._visible_shapes[i][1].z_index for i in inside_indices
+            ]
             pos = np.argsort(z_index)
-            return next(
-                inside_indices[p]
-                for p in pos[::-1]
-                if np.any(
-                    inside_triangles(
-                        self.shapes[inside_indices[p]]._all_triangles() - coord
+            return self._visible_shapes[
+                next(
+                    inside_indices[p]
+                    for p in pos[::-1]
+                    if np.any(
+                        inside_triangles(
+                            self._visible_shapes[inside_indices[p]][
+                                1
+                            ]._all_triangles()
+                            - coord
+                        )
```

Comment:
Have I mentioned I hate black? Not sure I have... 🤔 

😂 😭

Reply:
it's ok, we use ruff. 😜